### PR TITLE
Use new remote-store options in HA guides

### LINKED ISSUE #27508: Use new remote-store options in HA guides
### Description

Since #26756 there are new CLI options for the remote store, which will allow configuring remote Infinispan without a separate XML file.

### Discussion

_No response_

### Motivation

Simplify the installation.

### Details

A draft PR for KCB already exists in https://github.com/keycloak/keycloak-benchmark/pull/711

### PR DESCRIPTION
**Pull request by pruivo**
Closes #27508

Draft Note: It requires Infinispan 14.0.26.Final. I'll Infinispan in this PR too when it is released.

### CODE REVIEW

**Comment by ahus1** [comment]
@pruivo - I see you want to combine this with Infinispan 14.0.26.Final. 

My thoughts around that: The Infinispan 14.0.26.Final change is something we want to backport to KC24, the change of the CLI usage possibly not. If this is the case, it would be IMHO good to keep those two changes in separate PRs / issues.

**Comment by pruivo** [comment]
@ahus1 the 24.0 branch has the CLI options with the workaround:

https://github.com/keycloak/keycloak/blob/release/24.0/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/legacy/infinispan/CacheManagerFactory.java#L248

Don't you want to remove the workaround (and upgrade docs) when upgrading the Infinispan version? 

I'm okay with having a separate PR for the Infinispan upgrade though, but I think this should be in KC 24 at some point.

**Comment by ahus1** [comment]
@pruivo - ok, let me rephrase it. This is what I suggest: 

* One PR to upgrade Infinispan and remove the workaround from the docs and the Java code. This will then be backported.

* One PR to use the new CLI options in the docs. This will not be backported as is came quite late into the game.

Let me know if this clarifies it.

**Comment by pruivo** [comment]
@ahus1 updated with the documentation changes only. It uses the new CLI options everywhere and the code still enables the workaround.

**Comment by ahus1** [comment]
@pruivo - I added one more paragraph "Architecture" to explain what is happening in the background. I deemed this to be necessary as we now hide a lot what is going on in the CLI options. 

Please give it a look. If it looks good to you, I'll merge it. Thanks!

**Comment by pruivo** [comment]
@ahus1 I see, it makes sense and it looks good to me. Short and direct to the point :+1:

**Review by ahus1 — APPROVED** [review]
